src/components/CourseInfo/CourseInfo.jsx

```diff
@@ -9,9 +9,10 @@ import './CourseInfo.css';
 
 export default function CourseInfo() {
 	const { courseId } = useParams();
-	const courses = useSelector(selectCourses);
 	const authors = useSelector(selectAuthors);
-	const course = courses.find((course) => course.id === courseId);
+	const [course] = useSelector(selectCourses).filter(
+		(course) => course.id === courseId
+	);
```

Comment:
Could you please check, when I see add/update course page and click refresh I am redirected to courses page again while I should stay where I was, I guess this is because on page load you save only user name and do not save user role, that's why you are redirected. Also you need to request course info on page load. General advise: do not forget to test, reload pages and also check network tab to see the number of requests

Reply:
Right. It was because I tried to load role from state, not from localStorage. I added save role to localStorage on getUser and it will be removed on logout.